src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private int size;
+    private int threshold;
+    private List<Entry<K, V>>[] buckets;
+
+    public MyHashMap() {
+        this(INITIAL_CAPACITY);
+    }
+
+    public MyHashMap(int capacity) {
+        this.buckets = new List[capacity];
+        for (int i = 0; i < capacity; i++) {
+            buckets[i] = new ArrayList<>();
+        }
+        this.threshold = (int) (capacity * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        List<Entry<K, V>> bucket = buckets[index];
 
+        // Обробка ключа null
+        if (key == null) {
+            for (Entry<K, V> entry : bucket) {
+                if (entry.getKey() == null) {
+                    entry.setValue(value);
+                    return;
+                }
+            }
+            bucket.add(new Entry<>(null, value));
+            size++;
+            return;
+        }
+
+        for (Entry<K, V> entry : bucket) {
+            if (entry.getKey() != null && entry.getKey().equals(key)) {
+                entry.setValue(value);
+                return;
+            }
+        }
+        bucket.add(new Entry<>(key, value));
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        List<Entry<K, V>> bucket = buckets[index];
+
+        // Обробка ключа null
+        if (key == null) {
+            for (Entry<K, V> entry : bucket) {
+                if (entry.getKey() == null) {
+                    return entry.getValue();
```

Comment:
Don't use getters and setters in the inner class 'Entry' as you have direct access to its private fields. Replace 'entry.getValue()' with 'entry.value'.